docs/src/testing/using-cheatcodes.md

```diff
@@ -174,32 +197,81 @@ Failures:
-To achieve this, you need to precalculate address of the contract using `precalculate_address` of `ContractClassTrait` on declared contract,
-and then use it in `start_prank` as an argument:
-
+To `prank` the constructor, you need to `start_prank` before it is invoked, with the right address. To achieve this, you need to precalculate the address of the contract by using the `precalculate_address` function of `ContractClassTrait` on the declared contract, and then use it in `start_prank` as an argument:
 
 ```rust
 use snforge_std::{ declare, ContractClassTrait, start_prank };
 
 #[test]
-fn prank_the_constructor() {
+fn mock_constructor_with_prank() {
+    // declaring contract
     let contract = declare('HelloStarknet');
-    let constructor_arguments = @ArrayTrait::new();
-    
     // Precalculate the address to obtain the contract address before the constructor call (deploy) itself
-    let contract_address = contract.precalculate_address(constructor_arguments); 
-    
+    let contract_address = contract.precalculate_address(@ArrayTrait::new());
     // Change the caller address to 123 before the call to contract.deploy
     start_prank(contract_address, 123.try_into().unwrap());
-    
-    // The constructor will have 123 set as the caller address 
+
+    // The constructor will have 123 set as the caller address
     contract.deploy(constructor_arguments).unwrap();
 }
 ```
+
+### Mocking the constructor with `roll`
+
+Similarly, as we've seen with the `prank` cheatcode, you can also use the `roll` cheatcode to mock the constructor and pre-set the block number to a specific value, which will be stored in the `blk_number` variable. To do this, you'll need to use the `start_roll` cheatcode, which requires two input parameters: `contract_address` to specify the target contract and `block_number` to set the desired value.
+
+The `contract_address` can be derived by precalculating the address with `precalculate_address` function of theof `ContractClassTrait` on declared contract, as demonstrated in the `prank` example.
+
+It's important to note that the `start_roll` cheatcode needs to be used before deploying the contract to successfully mock the constructor.
+
+```rust
+#[test]
+fn mock_constructor_with_roll() {
+    // declaring contract
+    let contract = declare('HelloStarknet');
+    // precalculating the contract address
+    let contract_address = contract.precalculate_address(@ArrayTrait::new());
+    // set the block number to 234 for the precalculated address
+    start_roll(contract_address, 234);
+    // deploying contract
+    let contract_address = contract.deploy(@ArrayTrait::new()).unwrap();
+    // set dispatcher
+    let dispatcher = IHelloStarknetDispatcher { contract_address };
+    // retrieving the block number
+    let block_number = dispatcher.get_block_number();
+    // asserting if block number is 234 set by start_roll()
+    assert(block_number == 234, 'Wrong block number');
```

Comment:
The comments on this are a little excessive. Please concentrate on "why" instead of "what" when explaining the code to the user